### PR TITLE
Callback functionality for People API and openLink in App API

### PR DESCRIPTION
**Pull request by jadahiya-MSFT**
Added in TeamsJS v1 callback functionality for selectPeople and getContext.

### CODE REVIEW

**Comment by erinha** [comment]
<p>Analyzed commit id : d832ab015469988220eab8cdf10537d948fb0520</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: fab455cb346aff158f9ca0de677cc8355d21b421</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : ee04b3a5ad1df3462de22eae5060351ca0579e7b</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: fab455cb346aff158f9ca0de677cc8355d21b421</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : d93c98391f294b2d06e77fa5fcfdf0dc7e6649bb</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: fab455cb346aff158f9ca0de677cc8355d21b421</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : 530e095fada16c611e4eb4b7e6515e6a6950b47b</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.43 KB</td><td>59.42 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: bd112e44bd54dff4b1b231e9709ce6490a1b5076</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : 0565f39270fd3499315dea036107730692f62f97</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 126f401aae255673ebb77e31804e0ee6bb2cb2f5</p>

**Review by MarcinJuraszek — DISMISSED** [review]

**Review by Ella-ly — CHANGES_REQUESTED** [review]

File: apps/teams-test-app/src/components/PeopleAPIs.tsx [hunk 1]
```diff
@@ -14,9 +15,33 @@ const SelectPeople = (): React.ReactElement =>
...
+        withPromise: async input => {
+          const result = input ? await people.selectPeople(input) : people.selectPeople();
+          return JSON.stringify(result);
+        },
+        withCallback: (input, setResult) => {
+          const callback = (error: SdkError, people: people.PeoplePickerResult[]): void => {
+            if (error) {
+              setResult(JSON.stringify(error));
+            }
+            setResult(JSON.stringify(people));
```
[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply apps/teams-test-app/src/components/PeopleAPIs.tsx:35]: @Ella-ly I'm not sure I understand your code sample, but it looks like you have two identical else statements on the end. Do you just mean that the code should be updated to:

if (error) {
  setResult(JSON.stringify(error));
} else {
  setResult(JSON.stringify(people));
}

[the remark this answers is not in the document]
Comment (Ella-ly) [reply apps/teams-test-app/src/components/PeopleAPIs.tsx:35]: Yes. Sorry if it is confusing. The code in the comment description is the full representation, the **suggested change** part is where the code change can be made.

[the remark this answers is not in the document]
Comment (jadahiya-MSFT) [reply apps/teams-test-app/src/components/PeopleAPIs.tsx:35]: Whoops, my bad! Thanks Ella!

**Comment by erinha** [comment]
<p>Analyzed commit id : c1591acf06f23ba867691d897f6a7017f5fdc1f1</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 4e3f05ee3569953a280f1b7bba41332ef3db2200</p>

**Review by TrevorJoelHarris — DISMISSED** [review]
:shipit:

**Review by Ella-ly — DISMISSED** [review]

File: apps/teams-test-app/src/components/PeopleAPIs.tsx [hunk 2]
```diff
@@ -14,9 +15,34 @@ const SelectPeople = (): React.ReactElement =>
...
+            } else {
+              setResult(JSON.stringify(people));
+            }
+          };
+          if (input) {
+            people.selectPeople(callback, input);
+          } else {
+            people.selectPeople(callback);
+          }
+          return 'selectPeople()' + noHostSdkMsg;
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/PeopleAPIs.tsx:43]: Do we still need to return something here after your change to update how `withCallback` is defined (https://github.com/OfficeDev/microsoft-teams-library-js/commit/fa1cd710d0db4484bf970730e3908024a82f41fc)?

File: apps/teams-test-app/src/components/PeopleAPIs.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import { people } from '@microsoft/teams-js';
+import { people, SdkError } from '@microsoft/teams-js';
 import React, { ReactElement } from 'react';
 
+import { noHostSdkMsg } from '../App';
```
Comment (Ella-ly) [inline apps/teams-test-app/src/components/PeopleAPIs.tsx:4]: No `noHostSdkMsg` is used in this file.

File: apps/teams-test-app/src/components/AppAPIs.tsx [hunk 4]
```diff
@@ -31,9 +31,15 @@ const OpenLink = (): ReactElement =>
...
-      submit: async input => {
-        await app.openLink(input);
-        return 'Completed';
+      submit: {
+        withPromise: async input => {
+          await app.openLink(input);
+          return 'Completed';
+        },
+        withCallback: input => {
+          executeDeepLink(input);
```
Comment (Ella-ly) [inline apps/teams-test-app/src/components/AppAPIs.tsx:40]: I would suggest the code here to be:
```
       withCallback: (input, setResult) => {
          const onComplete = (status: boolean, reason?: string): void => {
            if (!status) {
              if (reason) {
                setResult(JSON.stringify(reason));
              } else {
                setResult("Status is false but there's not reason?! This shouldn't happen.");
              }
            } else {
              setResult('Completed');
            }
          };
          executeDeepLink(input, onComplete);
        },
```

File: apps/teams-test-app/src/components/PeopleAPIs.tsx (hunk 3 above)
Comment (jadahiya-MSFT) [reply apps/teams-test-app/src/components/PeopleAPIs.tsx:4]: good catch

**Review by Ella-ly — DISMISSED** [review]

**Review by Ella-ly — APPROVED** [review]

**Pull request merged**